reactor-test/src/main/java/reactor/test/publisher/TestPublisher.java

```diff
@@ -72,13 +75,43 @@
 	 * subscribers. It caches the {@link #next(Object)} events and replays them to
 	 * all subscribers upon subscription.
 	 * <p>
-	 * Note that this type of {@link Publisher} isn't
+	 * The returned publisher will buffer values if there is not enough request from subscribers.
 	 *
 	 * @param <T> the type of the publisher
 	 * @return the new {@link TestPublisher}
 	 */
 	public static <T> TestPublisher<T> createCold() {
-		return new ColdTestPublisher<>();
+		return new ColdTestPublisher<>(false, EnumSet.noneOf(Violation.class));
+	}
+
+	/**
+	 * Create a cold {@link TestPublisher}, which can be subscribed to by multiple
+	 * subscribers. It caches the {@link #next(Object)} events and replays them to
+	 * all subscribers upon subscription.
+	 * <p>
+	 * The returned publisher will emit an overflow error if there is not enough request from subscribers.
+	 *
+	 * @param <T> the type of the publisher
+	 * @return the new {@link TestPublisher}
+	 */
+	public static <T> TestPublisher<T> createColdNonBuffering() {
+		return new ColdTestPublisher<>(true, EnumSet.noneOf(Violation.class));
+	}
+
+	/**
+	 * Create a cold {@link TestPublisher}, which can be subscribed to by multiple
+	 * subscribers. It caches the {@link #next(Object)} events and replays them to
```

Comment:
it only caches if `boolean errorOnOverflow` is `false`

Reply:
No, it accumulates values and replays them to new subscribers in all cases. The boolean flag only drives what happens when there is more (total) request than values in the `values` buffer: wait for more request, or `error()`.

Reply:
ok then I'd rephrase the second paragraph in `createCold` and `createColdNonBuffering` above, then. Will comment there.